src/lib.rs

```diff
@@ -122,15 +125,32 @@ mod platform {
             let task = task_for_pid(self.pid);
             if task.is_err() { return task.map(|_| ()) }
 
-            unsafe {
-                let mut read: mach_msg_type_number_t = 0;
+            let page_addr      = (addr as i64 & (-4096)) as mach_vm_address_t;
+	        let last_page_addr = ((addr as i64 + buf.len() as i64 + 4095) & (-4096)) as mach_vm_address_t;
+            let page_size      = last_page_addr as usize - page_addr as usize;
```

Comment:
found some info:
- https://opensource.apple.com/source/xnu/xnu-517.3.7/osfmk/man/vm_read.html
- https://www.gnu.org/software/hurd/gnumach-doc/Data-Transfer.html#Data-Transfer

Looks like it must be a round number of pages. It also sounds like you should call `vm_deallocate` on the pointer you get back, or else leak the memory.

Kind of crappy that the interface requires copying, but what to do. There *is* a `vm_overwrite` that allows copying to a specified buffer, but since it has to be full pages we can't copy to the passed-in buffer anyway.

Looks